pandas/io/tests/test_parsers.py

```diff
@@ -2635,6 +2635,28 @@ def test_eof_states(self):
         self.assertRaises(Exception, self.read_csv,
                           StringIO(data), escapechar='\\')
 
+    def test_grow_boundary_at_cap(self):
+        # See gh-12494
+        #
+        # Cause of error was the fact that pandas
+        # was not increasing the buffer size when
+        # the desired space would fill the buffer
+        # to capacity, which later would cause a
+        # buffer overflow error when checking the
+        # EOF terminator of the CSV stream
+        def read_empty_header(count):
+            s = StringIO(',' * count)
+            df = read_csv(s)
+
+        for count in range(1, 101):
+            try:
+                read_empty_header(count)
+            except pandas.parser.CParserError as e:
```

Comment:
scratch that. Why are you catching an exception here at all? These should always be empty frames. In fact you can construct the expected frame and compare